e2e/e2e_test.go

```diff
@@ -136,6 +141,134 @@ func Test_No_Ethereum_TX_Submission(t *testing.T) {
+	var signatures [][]byte
+	for i := 0; i < len(transactionData.Signatures); i++ {
+		signature, _ := hex.DecodeString(transactionData.Signatures[i])
+		signatures = append(signatures, signature)
+	}
+
+	res, err := setupEnv.Clients.RouterContract.Mint(
+		setupEnv.Clients.KeyTransactor,
+		[]byte(fromHederaTransactionID(&transactionResponse.TransactionID).String()),
+		*tokenAddress,
+		common.HexToAddress(receiverAddress),
+		big.NewInt(int64(tokensSendAmount)),
+		signatures,
+	)
+
+	if err != nil {
+		t.Fatalf("Cannot execute transaction - Error: [%s].", err)
+	}
+	return res.Hash().String()
+}
+
+func waitForTransaction(setupEnv *setup.Setup, txHash string, t *testing.T) {
+	fmt.Printf("Waiting for transaction: [%s] to be mined\n", txHash)
+	c1 := make(chan bool, 1)
+	onSuccess := func() {
+		fmt.Printf("Transaction [%s] mined successfully\n", txHash)
+		c1 <- true
+	}
+	onRevert := func() {
+		t.Fatalf(`Failed to mine successfully ethereum transaction: [%s]`, txHash)
+	}
+	onError := func(err error) {
+		fmt.Println(fmt.Sprintf(`Transaction unsuccessful, Error: [%s]`, err))
+		t.Fatal(err)
+	}
+	setupEnv.Clients.EthClient.WaitForTransaction(txHash, onSuccess, onRevert, onError)
+	<-c1
+}
+
+func validateTokenBalance(setupEnv *setup.Setup, wrappedTokenBalanceBefore *big.Int, wTokenReceiverAddress common.Address, t *testing.T) {
+	wrappedTokenBalanceAfter, err := setupEnv.Clients.WTokenContract.BalanceOf(&bind.CallOpts{}, wTokenReceiverAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tokensAmount := new(big.Int).SetInt64(int64(tokensSendAmount))
+
+	serviceFeePercentage, err := setupEnv.Clients.RouterContract.ServiceFee(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	txFee := new(big.Int).Mul(tokensAmount, serviceFeePercentage)
+	txFee = new(big.Int).Div(txFee, precision)
+
+	expectedBalance := new(big.Int).Sub(wrappedTokenBalanceAfter, wrappedTokenBalanceBefore)
+	mintAmount := new(big.Int).Sub(tokensAmount, txFee)
+
+	if expectedBalance.Cmp(mintAmount) != 0 {
+		t.Fatalf("Incorect token balance")
+	}
+}
+
+func verifyTransferFromValidatorAPI(setupEnv *setup.Setup, txResponce hedera.TransactionResponse, signatures []string, t *testing.T) (*service.TransferData, *common.Address) {
+	tokenAddress, _ := setup.ParseToken(setupEnv.Clients.RouterContract, setupEnv.TokenID.String())
+
+	transactionData, err := setupEnv.Clients.ValidatorClient.GetTransferData(fromHederaTransactionID(&txResponce.TransactionID).String())
+	if err != nil {
+		t.Fatalf("Cannot fetch transaction data - Error: [%s].", err)
+	}
+	if transactionData.Amount != fmt.Sprint(tokensSendAmount) {
+		t.Fatal("Transaction data mismatch")
+	}
+	if transactionData.NativeToken != setupEnv.TokenID.String() {
+		t.Fatal("Native Token mismatch")
+	}
+	if transactionData.Recipient != receiverAddress {
+		t.Fatal("Receiver address mismatch")
+	}
+	if transactionData.WrappedToken != tokenAddress.String() {
+		t.Fatal("Token address mismatch")
+	}
```

Comment:
Can we make the fatal logs a bit more descriptive, similar to this suggestion?
`Incorrect token balance. Expected to be [%s], but was [%s].`